File: backend/src/models/StudentProfile.model.js
import mongoose from 'mongoose';

const studentProfileSchema = new mongoose.Schema({

    userId: { 
        type: mongoose.Schema.Types.ObjectId, /// It was ObjectId
        ref: 'User',
        required: true,
        unique: true
    },

    rollNo: {
        type: String,
        required: true,
        unique: true,
        trim: true,
        uppercase: true,
    },

    stdName: {
        type: String,
        required: true,
        trim: true,
    },


    fatherName: {
        type: String,
        required: true,
        trim: true,
    },

    fatherPhone: {
        type: String,
        required: true,
        match: [/^[0-9]{11}$/, 'Valid 11-digit phone number'],
    },

    contact: {
        type: String
    },

    address: {
        type: String,
        required: true
    },

    age: {
        type: Number,
        required: true
    },

    className: {
        type: String,
        required: true,
        uppercase: true,
        trim: true,
    },

    field: {
        type: String,
        enum: ['cs', 'medical', 'engineering', 'commerce', 'arts'],  // 
        required: true,
    },
}, { timestamps: true });

// studentProfileSchema.index({ rollNo: 1 }); // Roll No ke basis pr fast searching ke liye
// studentProfileSchema.index({ className: 1 }); // Class wise filtering ke liye

export default mongoose.model('StudentProfile', studentProfileSchema);

File: MobileApp/irp_mobile_app/lib/models/student_profile_model.dart
class StudentProfileModel {
  final String id;
  final String userId;
  final String rollNo;
  final String stdName;
  final String fatherName;
  final String fatherPhone;
  final String? contact;
  final String address;
  final int age;
  final String className;
  final String field;

  StudentProfileModel({
    required this.id,
    required this.userId,
    required this.rollNo,
    required this.stdName,
    required this.fatherName,
    required this.fatherPhone,
    this.contact,
    required this.address,
    required this.age,
    required this.className,
    required this.field,
  });

  factory StudentProfileModel.fromJson(Map<String, dynamic> json) {
    return StudentProfileModel(
      id: json['_id'] ?? '',
      userId: json['userId'] ?? '',
      rollNo: json['rollNo'] ?? '',
      stdName: json['stdName'] ?? '',
      fatherName: json['fatherName'] ?? '',
      fatherPhone: json['fatherPhone'] ?? '',
      contact: json['contact'],
      address: json['address'] ?? '',
      age: json['age'] ?? 0,
      className: json['className'] ?? '',
      field: json['field'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'userId': userId,
      'rollNo': rollNo,
      'stdName': stdName,
      'fatherName': fatherName,
      'fatherPhone': fatherPhone,
      'contact': contact,
      'address': address,
      'age': age,
      'className': className,
      'field': field,
    };
  }
}
